### PR TITLE
Checklist Analytic Receiver

### PR DESCRIPTION
**Pull request by sdub18**
Should cover the checklist analytic. Please let me know if I missed anything by mistake!

### CODE REVIEW

File: AppBookDataAnalyzer/Analytics for Rows/ChecklistNumItemsTapped.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            
+            switch textbookMaterial {
+                
+            case .page(let appbook, let pageNumber):
+                
+                let query = """
+                    SELECT COUNT(*)
+                    FROM \(Database.ChecklistSelectedItems.tableName)
+                    JOIN \(Database.ChecklistItems.tableName)
+                    ON \(Database.ChecklistItems.Column.id) = \(Database.ChecklistSelectedItems.Column.checklistItemId)
```
Comment (jeremy6462) [inline AppBookDataAnalyzer/Analytics for Rows/ChecklistNumItemsTapped.swift:26]: For clarity, this should be `ON \(Database.ChecklistItems.tableName).\(Database.ChecklistItems.Column.id) = \(Database.ChecklistSelectedItems.tableName).\(Database.ChecklistSelectedItems.Column.checklistItemId)`. This means that we'll join on `checklist_items.id = checklist_selected_items.checklist_item_id` which is more specific than what is written now: `id = checklist_item_id`.

File: AppBookDataAnalyzer/Analytics for Rows/ChecklistNumItemsTapped.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                    JOIN \(Database.QuestionPageJoin.tableName)
+                    ON \(Database.QuestionPageJoin.Column.id) = \(Database.QuestionPageJoin.Column.questionId)
```
Comment (jeremy6462) [inline AppBookDataAnalyzer/Analytics for Rows/ChecklistNumItemsTapped.swift:28]: Similarly to above, we should include the table names here for clarity. Moreover, we shouldn't join on two columns of the same table (you're joining on `question_to_page_join.id = question_to_page_join.question_id` here, when really it should be `question_to_page_join.id = checklist_items.question_id`

Comment (jeremy6462) [inline AppBookDataAnalyzer/Analytics for Rows/ChecklistNumItemsTapped.swift:28]: Similarly to above, we should include the table names here for clarity. Moreover, we shouldn't join on two columns of the same table (you're joining on `question_to_page_join.id = question_to_page_join.question_id` here, when really it should be `question_to_page_join.id = checklist_items.question_id`

**Review by jeremy6462 — CHANGES_REQUESTED** [review]
Thanks for adding this in! While the query is correct, I have some nitpicks for clarity. All good after that!

**Review by fjfur10 — APPROVED** [review]
Looks good to me, you may want to remove the `.xcuserstate` before merging though (the file should be added to `.gitignore` in `main`)

**Comment by jeremy6462** [comment]
> Looks good to me, you may want to remove the `.xcuserstate` before merging though (the file should be added to `.gitignore` in `main`)

I think if you merge main into this branch that should remove the file

**Comment by sdub18** [comment]
@jeremy6462 Could you give this PR another run through? I believe I fixed the issues with the SQL query

**Review by jeremy6462 — APPROVED** [review]
All good from my end now!

**Pull request merged**